### PR TITLE
New-DbaDbUser only working on first database #6052

### LINKED ISSUE #6052: [Bug]New-DbaDBUser only works with first database and then throws an error
### Environmental information

```
#### PowerShell version:

Name                           Value                                                                                   
----                           -----                                                                                   
PSVersion                      5.1.17134.858                                                                           
PSEdition                      Desktop                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
BuildVersion                   10.0.17134.858                                                                          
CLRVersion                     4.0.30319.42000                                                                         
WSManStackVersion              3.0                                                                                     
PSRemotingProtocolVersion      2.3                                                                                     
SerializationVersion           1.1.0.1                                                                                 




#### dbatools Module version:


Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\1.0.46\dbatools.psd1
Version : 1.0.46

Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.754\dbatools.psd1
Version : 0.9.754

Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.734\dbatools.psd1
Version : 0.9.734

Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.301\dbatools.psd1
Version : 0.9.301

Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.209\dbatools.psd1
Version : 0.9.209


#### PowerShell version:

Name                           Value                                                                                              
----                           -----                                                                                              
PSVersion                      4.0                                                                                                
WSManStackVersion              3.0                                                                                                
SerializationVersion           1.1.0.1                                                                                            
CLRVersion                     4.0.30319.42000                                                                                    
BuildVersion                   6.3.9600.19170                                                                                     
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0}                                                                               
PSRemotingProtocolVersion      2.2                                                                                                




#### dbatools Module version:


Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\dbatools.psd1
Version : 1.0.46


```
#### SQL Server: 

```sql
Microsoft SQL Server 2017 (RTM-GDR) (KB4505224) - 14.0.2027.2 (X64) 
	Jun 15 2019 00:26:19 
	Copyright (C) 2017 Microsoft Corporation
	Developer Edition (64-bit) on Windows 10 Pro 10.0 <X64> (Build 17134: ) (Hypervisor)

Microsoft SQL Server 2012 (SP4) (KB4018073) - 11.0.7001.0 (X64) 
	Aug 15 2017 10:23:29 
	Copyright (c) Microsoft Corporation
	Standard Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: )

```

### Report

When running New-DbaDBUser the cmdlet runs and adds the user to the first database and then throws this warning:

` WARNING: [19:21:36][New-DbaDbUser] Invalid Login: [LoginName] is not found on [ServerName]  `

If I pass the -EnableException this is the error:

```
Invalid Login: [LoginName] is not found on [Servername]
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.0.46\allcommands.ps1:81279 char:9
+         throw $records[0]
+         ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (localhost:String) [], Exception
    + FullyQualifiedErrorId : dbatools_New-DbaDbUser
```

I have run this against SQL 2017 and SQL 2012.  In both cases the user I am executing it as is sysadmin on the SQL Server.  I have run it in the ISE,  using powershell.exe, and in the console all with the same results.

 

#### Host used

- [X ] powershell.exe
- [X ] ISE
- [ ] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [X ] Still buggy in powershell.exe

#### Errors Received

```


PSMessageDetails      : 
Exception             : System.Exception: Invalid Login: [NewLogin] is not found on [ServerName]
TargetObject          : localhost
CategoryInfo          : NotSpecified: (localhost:String) [], Exception
FullyQualifiedErrorId : dbatools_New-DbaDbUser
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Stop-Function, C:\Program Files\WindowsPowerShell\Modules\dbatools\1.0.46\allcommands.ps1: 
                        line 81279
                        at New-DbaDbUser<Process>, C:\Program 
                        Files\WindowsPowerShell\Modules\dbatools\1.0.46\allcommands.ps1: line 52244
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}

```
#### Steps to Reproduce

I have also tried a specific foreach loop like this:

```
Remove-DbaLogin -SqlInstance localhost -Login NewLogin

New-DbaLogin -SqlInstance localhost -Login NewLogin 

foreach($database in (Get-DBaDatabase -SqlInstance localhost -ExcludeSystem |Select SqlInstance, Name))
{
    New-DbaDBUser -SqlInstance $database.SqlInstance -Database $database.Name -Login NewLogin -Confirm -Force -EnableException
}
```

This works one time in the loop and then fails with the same error.  Here is the output:

```
ComputerName : [ComputerName]
InstanceName : MSSQLSERVER
SqlInstance  : [SqlInstance]
Login        : NewLogin
Status       : Dropped

ComputerName : [ComputerName]
InstanceName : MSSQLSERVER
SqlInstance  : [SqlInstance]
Name         : NewLogin
LoginType    : SqlLogin
CreateDate   : 9/19/2019 7:47:03 PM
LastLogin    : 
HasAccess    : True
IsLocked     : False
IsDisabled   : False

ComputerName : [ComputerName]
InstanceName : MSSQLSERVER
SqlInstance  : [SqlInstance]
Database           : [DBName]
CreateDate         : 9/19/2019 7:47:10 PM
DateLastModified   : 9/19/2019 7:47:10 PM
Name               : NewLogin
Login              : NewLogin
LoginType          : SqlLogin
AuthenticationType : Instance
State              : Existing
HasDbAccess        : True
DefaultSchema      : dbo

Error occurred while establishing a connection to [NextDB]
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.0.46\allcommands.ps1:81279 char:9
+         throw $records[0]
+         ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ConnectionError: ([ServerName]:String) [], Exception
    + FullyQualifiedErrorId : dbatools_New-DbaDbUser

```
#### Expected Behavior
See the login dropped, recreated, then a user created in each database on the SQL Server for the login

#### Actual Behavior
Worked in the first database encountered and then failed.

### PR DESCRIPTION
**Pull request by richardlhughes**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #6052)
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
Fixes reported bug New-DbaDbUser only working on first database #6052 
### Approach

The parameter for login was getting set incorrectly at the end of the foreach loop causing only the first database to work. 

### Commands to test

New-DbaLogin -SqlInstance localhost -Login $login -SqlCredential $cred
New-DbaDbUser -SqlInstance localhost -Login $login -Confirm -Force -EnableException -Verbose -Debug -SqlCredential $cred

### CODE REVIEW

**Comment by wsmelton** [comment]
Would you be able to update the test for this command as well, to validate that your change addresses it and prevents it from happening in future code changes?

**Comment by richardlhughes** [comment]
Yes, let me work on getting to this part.

Thanks again for helping me on the slack channel and getting the details on setting up the testing.

File: tests/New-DbaDbUser.Tests.ps1 [hunk 1]
```diff
@@ -12,7 +12,7 @@ Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
         }
     }
     Context "Should error out if the database does not exist" {
-        Mock -CommandName Connect-SqlInstance {
+        Mock -CommandName Connect-DbaInstance {
```
Comment (wsmelton) [inline tests/New-DbaDbUser.Tests.ps1]: Please revert this back, the `Connect-SqlInstance` is the function used in `New-DbaDbUser`.

**Comment by wsmelton** [comment]
Test run against current gallery build of module:
image

Test run against this PR:
image

**Comment by wsmelton** [comment]
Hopefully we get a clean run from Appveyor and then this can be merged.

**Comment by potatoqualitee** [comment]
Ill try pushing it, @wsmelton

**Comment by potatoqualitee** [comment]
oh wait, all checks have passed ✔️  will merge! thank you both

**Pull request merged**